tools/signingtool/src/test/java/com/horizen/SigningToolTest.java

```diff
@@ -0,0 +1,172 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Random;
+
+import static junit.framework.TestCase.*;
+
+
+public class SigningToolTest {
+
+    public static final String SCHNORR = "schnorr";
+    private ByteArrayOutputStream byteArrayOutputStream;
+    private PrintStream console;
+    public static final SidechainSecretsCompanion SECRETS_COMPANION = new SidechainSecretsCompanion(new HashMap<>());
+    public static final SchnorrPropositionSerializer PROPOSITION_SERIALIZER = SchnorrPropositionSerializer.getSerializer();
+
+    @Before
+    public void setup() {
+        byteArrayOutputStream = new ByteArrayOutputStream();
+        console = System.out;
+    }
+
+    private String runTest(final String[] args) {
+        try {
+            System.setOut(new PrintStream(byteArrayOutputStream));
+            SigningTool.main(args);
+            return byteArrayOutputStream.toString();
+        } catch (Exception e) {
+            fail("Unexpected error in tests: " + e.getMessage());
+        } finally {
+            System.setOut(console);
+        }
+        return null;
+    }
+
+    @Test
+    public void testUsage() {
+        String command = "help";
+        String result = runTest(new String[]{command});
+        assertTrue(result != null && result.contains("Usage") && result.contains("Supported commands"));
+    }
+
+    @Test
+    public void testSignatureSignVerify() throws JsonProcessingException {
+        //Sign
+        String command = "createSignature";
+        String testMessage = "Test message to sign/verify.";
+        SchnorrSecret secretKey = generateSecret();
+        SchnorrProposition publicKey = secretKey.publicImage();
+
+        ObjectNode argsJson = new ObjectMapper().createObjectNode();
+        argsJson.put("message", testMessage);
+        argsJson.put("privateKey", BytesUtils.toHexString(SECRETS_COMPANION.toBytes(secretKey)));
+        argsJson.put("type", SCHNORR);
+        String args = argsJson.toString();
+
+        String result = runTest(new String[]{command, args});
+        assertTrue(result != null && result.contains("signature") && !result.contains("error"));
+
+        //Verify
+        command = "verifySignature";
+        JsonNode signatureJson = new ObjectMapper().readTree(result);
+        String signature = signatureJson.get("signature").textValue();
+        argsJson = new ObjectMapper().createObjectNode();
+        argsJson.put("message", testMessage);
+        argsJson.put("signature", signature);
+        argsJson.put("publicKey", BytesUtils.toHexString(PROPOSITION_SERIALIZER.toBytes(publicKey)));
+        argsJson.put("type", SCHNORR);
+        args = argsJson.toString();
+        result = runTest(new String[]{command, args});
+        assertTrue(result != null && result.contains("valid") && result.contains("true"));
+
+        //Negative
```

Comment:
add tests with invalid signature and invalid public keys. Also signature or key of invalid types (that can't be even deserialized)
Same in other tests